File: Code/Backend/Backend/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models;

/// <summary>
/// 收货地址表 — 用户的收货信息
/// </summary>
[Table("address")]
public class Address
{
    [Key]
    [Column("address_id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int AddressId { get; set; }

    [Column("name")]
    [MaxLength(10)]
    public string Name { get; set; } = string.Empty;

    [Column("phone_number")]
    [MaxLength(20)]
    public string PhoneNumber { get; set; } = string.Empty;

    [Column("detail_address")]
    [MaxLength(50)]
    public string DetailAddress { get; set; } = string.Empty;

    [Column("user_id")]
    public int UserId { get; set; }

    /// <summary>
    /// 是否默认地址：0=否，1=是
    /// </summary>
    [Column("is_default")]
    public int IsDefault { get; set; } = 0;

    // 导航属性
    [ForeignKey("UserId")]
    public NormUser? User { get; set; }
}

File: Code/Frontend/src/types/api/address.ts
/**
 * 地址响应 DTO
 */
export interface AddressDto {
    addressId: number
    name: string
    phoneNumber: string
    detailAddress: string
    isDefault: boolean
}

/**
 * 新增地址请求
 */
export interface CreateAddressRequest {
    name: string
    phoneNumber: string
    detailAddress: string
    isDefault: boolean
}

/**
 * 修改地址请求（部分更新，字段为 null 时不修改）
 */
export interface UpdateAddressRequest {
    name?: string | null
    phoneNumber?: string | null
    detailAddress?: string | null
}
